### PR TITLE
Fix Issue 11455 - Overriding template methods should raise a compile …

### PR DESCRIPTION
**Pull request by ntrel**
…error

@RazvanN7 This implements https://github.com/orgs/dlang/projects/9/views/1?pane=issue&itemId=10645246

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by RazvanN7** [comment]
This seems to break Vectorflow:

```


src/vectorflow/layers.d(476,19): Error: a function template cannot be `override`
--
  | src/vectorflow/neurallayer.d(243,19): Error: a function template cannot be `override`
  | src/vectorflow/neurallayer.d(246,19): Error: a function template cannot be `override`

```

File: compiler/src/dmd/parse.d [hunk 1]
```diff
@@ -4596,6 +4596,11 @@ class Parser(AST, Lexer = dmd.lexer.Lexer) : Lexer
                  */
                 if (tpl)
                 {
+                    if (storage_class & STC.override_)
+                        error(loc, "a function template cannot be `override`");
```
Comment (RazvanN7) [inline compiler/src/dmd/parse.d:4600]: Maybe a bit more info would come in handy. Something along the lines of: "Function templates are never virtual, therefore they cannot override anything." and then add an error supplemental stating: "`override` used to be accepted and ignored on function templates. Please delete the `override` keyword."

**Comment by RazvanN7** [comment]
Also, maybe we should use a deprecation first. Vectorflow looks unmaintained, I'll raise a PR to fix the failure, but chances are it's gonna get ignored.

**Comment by RazvanN7** [comment]
PR: https://github.com/Netflix/vectorflow/pull/35

**Pull request merged**